### PR TITLE
update cloudevents/sdk-go to v2.0.0preview3

### PR DESCRIPTION
**Pull request by nlopezgi**
## Proposed Changes

- Update to latest release of cloudevents/sdk-go after https://github.com/knative/eventing/pull/2764
- github.com/cloudevents/sdk-go/pkg/cloudevents -> github.com/cloudevents/sdk-go/legacy/pkg/cloudevents

### CODE REVIEW

**Comment by nlopezgi** [comment]
had to ignore a couple of fields in adapter_test.go to get unit tests passing, ptal.

File: pkg/pubsub/adapter/adapter_test.go [hunk 1]
```diff
@@ -197,9 +196,6 @@ func TestInboundConvert(t *testing.T) {
 }
 
 func TestReceive(t *testing.T) {
-	traceID := "1234567890abcdef1234567890abcdef"
```
Comment (nlopezgi) [inline pkg/pubsub/adapter/adapter_test.go]: fyi @ian-mi @nachocano Had to ignore these fields to get tests passing. Let me know if there is a better alternative.

Comment (nachocano) [reply pkg/pubsub/adapter/adapter_test.go]: maybe just comment this out instead of removing and add a TODO to the issue will or has created on the SDK?

Comment (nlopezgi) [reply pkg/pubsub/adapter/adapter_test.go]: done

**Comment by nachocano** [comment]
/lgtm

Thanks for the quick turnaround @nlopezgi !!!

**Pull request merged**